### PR TITLE
Clear the AS (alignment score) tag when REMOVE_ALIGNMENT_INFORMATION …

### PR DESCRIPTION
**Pull request by ktibbett**
…is true.

### CODE REVIEW

File: src/java/picard/sam/RevertSam.java [hunk 1]
```diff
@@ -374,6 +374,7 @@ public void revertSamRecord(final SAMRecord rec) {
             rec.setAlignmentStart(SAMRecord.NO_ALIGNMENT_START);
             rec.setCigarString(SAMRecord.NO_ALIGNMENT_CIGAR);
             rec.setMappingQuality(SAMRecord.NO_MAPPING_QUALITY);
+            rec.setAttribute(SAMTag.AS.name(), null);
```
Comment (nh13) [inline src/java/picard/sam/RevertSam.java]: What about other alignment based tags, such as ```MD```, ```MC```, or ```NM```?   Why not just add ```AS``` to ```ATTRIBUTE_TO_CLEAR```?

**Comment by ktibbett** [comment]
@nh13 -- thanks.  Ok now?

**Comment by sooheelee** [comment]
FWI, here are all the tags BWA uses:

NM, MD, AS, BC, X0, X1, XN, XM, XO, XG, XT, XA, XS, XF, XE

Soo Hee Lee, Ph.D.
Science Technical Writer

**Comment by sooheelee** [comment]
So better yet, could we get an ALL option that clears all queryname-associated tags?

**Comment by nh13** [comment]
@ktibbett looks great. @sooheelee I would recommend not adding all the ```X?``` tags to the default list as those are not standardized across aligners and tools.  The user can use the ```ATTRIBUTE_TO_CLEAR ``` command line option to add those.

**Comment by ktibbett** [comment]
@sooheelee , BC is not alignment-releated, and the other tags are custom to the tool vendor.  I agree with @nh13  that the ATTRIBUTE_TO_CLEAR argument gives users sufficient control over what other tags they do and do not retain.

**Comment by sooheelee** [comment]
@nh13 , I will disagree. I think in the least we should provide an option to clear all tags that will subsequently be produced by our GATK variant discovery workflow. This way users can start with a clean file without having to learn what I spent a considerable amount of time on yesterday, that is the command to comprehensively list all tags present in a BAM file. It turns out the BAM given to me by aggregation also contained OC and OP tags that were then misapplied back to the BAM during the MergeBamAlignment step. I am assuming here that the OC and OP tags are applied to a BAM during one of our processing steps downstream of where I am.

**Comment by nh13** [comment]
Picard is a community set of tools, and not just for Broad's Picard+GATK variant discovery workflow.  For tags introduced by that latter workflow, perhaps the Broad can document those in the GATK forums and list the tags needed to clean the BAM.

**Comment by sooheelee** [comment]
@nh13 I'll keep that in mind for our documentation. As for my comment, I am doing my bit as a voice of dissension on behalf of users.

**Comment by nh13** [comment]
@sooheelee I completely understand and keep it up!

**Pull request merged**